source/auth/auth.rst

```diff
@@ -1177,76 +1177,56 @@ If AWS authentication fails for any reason, the cache MUST be cleared.
 
-Note that the principal name is optional as it may be provided by the IDP in environments where only one IDP is used.  It is given by the user as the
-username.
+Note that the principal name is optional as it may be provided by the IDP in environments where only one IDP is used.  The username provided by the user MUST be used as the principalName.
 
 .. code:: idl
 
-  OIDCMechanismServerStep1:
-    description: "Server's reply to clientStep1"
-    strict: false
+  IdPServerInfo:
+    description: "The information used by callbacks to authenticate with the Identity Provider."
     fields:
-      authorizationEndpoint:
-        description: >-
-          URL where the IDP may be contacted for end user
-          authentication and authorization code generation.
-        type: string
-        optional: true # Req if deviceAuthorizeEndpoint not present
-      tokenEndpoint:
+      issuer:
         description: >-
-          URL where the IDP may be contacted for authorization
-          code <=> ID/access token exchange.
-        type: string
-        optional: true # Req if deviceAuthorizeEndpoint not present
-      deviceAuthorizationEndpoint:
-        description: >-
-          URL where the IDP may be contacted for device
-          authentication and authorization code generation.
-        type: string
-        optional: true # Req if authorizeEndpoint not present
+            URL which describes the Authentication Server. This identifier should be
+            the iss of provided access tokens, and be viable for RFC8414
+            metadata discovery and RFC9207 identification.
+        type:string
       clientId:
         description: "Unique client ID for this OIDC client"
         type: string
-      clientSecret:
-        description: "Secret used when communicating with IDP"
-        type: string
-        optional: true
       requestScopes:
         description: "Additional scopes to request from IDP"
         type: array<string>
         optional: true
 
-Server will use principalName (n) if provided in clientStep1 to select an appropriate IDP.  This IDP's configuration will be returned in serverStep1 to instruct the client on how to acquire an Access Token.  This Access Token will be used in clientStep2 to complete authentication.  The principalName identified within the Access Token MUST match any provided during clientStep1, or the authentication operation will fail.
+Server will use principalName (n) if provided in the driver’s PrincipalStepRequest to select an appropriate IDP.  This IDP's configuration will be returned in the server’s response that will be used by the end-user to acquire an Access Token.
+
+This Access Token will be used as the JWT in the driver’s JwtStepRequest to complete authentication. Users implementing callbacks must ensure that the principalName identified within the Access Token must match the principalName provided to the callback, otherwise the JwtStepRequest will fail on the server.
```

Comment:
[non-blocking, can be clarified later] Given that the callbacks do not receive the principalName directly anymore, I'm not sure how much sense the clarification in the last sentence of this paragraph makes anymore

Reply:
Removed